src/CacheManager.Core/ICacheManager.cs

```diff
@@ -611,6 +611,28 @@ public interface ICacheManager<TCacheValue> : ICache<TCacheValue>
         bool TryUpdate(string key, string region, Func<TCacheValue, TCacheValue> updateValue, int maxRetries, out TCacheValue value);
 
         /// <summary>
+        /// Gets all keys that match the given pattern.
```

Comment:
protected GetKeys on BaseCacheHandle can't be called from BaseCacheManager, and making them abstract would been implementing the exceptions in every cache layer, even if they don't support Keys. I implemented just "SupportsKeys" as abstract and the rest as virtuals that throw NotImplemented by default. Does the implementation meet your requirements?